### ISSUE TITLE
[🚀] Make individual challenges "reset-able" & always show hint/tutorial buttons

### ISSUE THREAD

**Issue by CrazyKidJack** [juice-shop/juice-shop#1784]
# :rocket: Feature request

### Description

 A clear and concise description of the problem or missing capability...
When solving challenges, sometimes you believe there may be multiple methods or perhaps you just "accidentally" solved a challenge. The problem is that once the challenge is marked complete, you can no longer see the hint, click the hint link (though you could go to it manually), or access the tutorial (if one exists).

### Solution ideas

 If you have a solution in mind, please describe it.

1. Make individual challenges "reset-able"
2. Don't hide any of the buttons (like help or tutorial) once a challenge is solved

### Possible alternatives

 Have you considered any alternative solutions or workarounds?
I use a docker container so...

1. Save progress
2. Stop application
3. Completely clear all browser cookies
4. Start application
5. Re-try challenge / access buttons that you want to access
6. Once finished with that challenge, reload progress

**Comment by bkimminich** [comment]
I think there's really just three challenges which you _often_ solve "by accident" or as a "side effect":
* _Privacy Policy_ (because you solve it by just visiting that openly accessible page)
* _Error Handling_ (by causing literally _any_ error that falls through onto the generic error page)
* _Basket Access_ (because your browser cache might contain something from a previous login when logging in with a different user)

If this happens to you with other challenges, please list them here along with the situation where they triggered unexpectedly. This might be interesting to tune the anti-cheat mechanism as well.

But as for resetting challenges, I'm not convinced that this is worth the negative side effects of squeezing yet another button onto the Score Board. Same for keeping the hint/tutorial button, this would be two more buttons that eat up precious space and they are not needed in probably 97% of all cases. Launching tutorials would be even dangerous, because they might no longer work when the challenge was already solved, as some preconditions have already changed. Again, causes more trouble than benefit if you ask me.

Like #1785 I'll mark this one with `wontfix` but leave it open for a while in case others find it (not) useful and want to chime in on the argument for (or against) implementing this.

**Comment by CrazyKidJack** [comment]
I think I didn't communicate clearly. Challenges are being marked as completed "unexpectedly". It's just that... sometimes the hint will say "there are 3 options". You do one and it works but you don't remember the other two. If you want to see them again, you either have to find it in the hint pdf (which isn't very easy imo partially because you can't {ctl} + f) or you have to restart the application.

The other thing that happens is, you're working on a challenge, you're trying stuff... taking notes... and then something works... but you don't really understand why and in some cases you aren't even sure that what you did worked without the confetti and banner (partially because you are still learning and don't understand what you did). so what I want to do as a student is try it again and see if i can re-trigger the banner to indicate that I am able to duplicate my success. Again, this is only possible be restarting the application

Basically I just want to be able to try the challenges (with success feedback) multiple times

**Comment by bkimminich** [comment]
I'm closing this one because it did not get any traction. The existing Score Board is already quite inflated with buttons and information, a reset button would only make this worse. Also it would need to be handled in a way that people couldn't easily reset progress for instances of others in e.g. CTFs. Might be re-evaulated if a new Score Board is ever being designed.
